src/sundials/sundials_iterative.c

```diff
@@ -324,22 +324,24 @@ int SUNQRsol(int n, sunrealtype **h, sunrealtype *q, sunrealtype *b)
  * -----------------------------------------------------------------
  */
 
-int SUNQRAdd_MGS(N_Vector *Q, sunrealtype *R, N_Vector df,
-                 int m, int mMax, void *QRdata)
+SUNErrCode SUNQRAdd_MGS(N_Vector* Q, sunrealtype* R, N_Vector df, int m, int mMax,
+                        void* QRdata)
 {
-    sunindextype j;
-    SUNQRData qrdata = (SUNQRData) QRdata;
-
-    N_VScale(ONE, df, qrdata->vtemp);
-    for (j=0; j < m; j++) {
-      R[m * mMax + j] = N_VDotProd(Q[j], qrdata->vtemp);
-      N_VLinearSum(ONE, qrdata->vtemp, -R[m * mMax + j], Q[j], qrdata->vtemp);
-    }
-    R[m * mMax + m] = SUNRsqrt(N_VDotProd(qrdata->vtemp, qrdata->vtemp));
-    N_VScale((1/R[m * mMax + m]), qrdata->vtemp, Q[m]);
+  sunindextype j;
+  SUNQRData qrdata  = (SUNQRData)QRdata;
+  SUNFunctionBegin(Q[0]->sunctx);
```

Comment:
This case isn't a good example but `SUNFunctionBegin` would need to go first for functions using `SUNAssert` for input checking and then using those inputs to initialize variables when declaring them. As a general rule it might be easier to always have `SUNFunctionBegin` first or are there cases where this could be an issue?